packages/aws-rfdk/lib/deadline/test/worker-fleet.test.ts

```diff
@@ -1461,6 +1469,36 @@ describe('HealthMonitor Tests', () => {
       TargetType: 'instance',
     }));
   });
+
+  test('UserData is added', () => {
+    // WHEN
+    class UserDataProvider extends InstanceUserDataProvider {
+      preCloudWatchAgent(host: IHost): void {
+        host.userData.addCommands('echo preCloudWatchAgent');
+      }
+      preDeadlineConfiguration(host: IHost): void {
+        host.userData.addCommands('echo preDeadlineConfiguration');
+      }
+      postWorkerLaunch(host: IHost): void {
+        host.userData.addCommands('echo postWorkerLaunch');
+      }
+    }
+    const fleet = new WorkerInstanceFleet(wfstack, 'workerFleet', {
+      vpc,
+      workerMachineImage: new GenericWindowsImage({
+        'us-east-1': 'ami-any',
+      }),
+      renderQueue,
+      healthMonitor,
+      userDataProvider: new UserDataProvider(),
+    });
+    const userData = fleet.fleet.userData.render();
+
+    // THEN
+    expect(userData).toContain('echo preCloudWatchAgent');
+    expect(userData).toContain('echo preDeadlineConfiguration');
+    expect(userData).toContain('echo postWorkerLaunch');
```

Comment:
These make no guarantees that the user data are injected in the correct places. Any ideas on how we can test that here?